test/conformance/conformance_parallel_pipeline.cpp

```diff
@@ -331,18 +331,17 @@ TEST_CASE("Testing max token number")
     for(unsigned int i = 1; i < n_tokens; i++)
     {
         std::atomic<short> active_tokens{0};
-        int counter{0};
 
         oneapi::tbb::filter<void,int> filter1(oneapi::tbb::filter_mode::parallel,
-                                [&active_tokens,&counter](oneapi::tbb::flow_control&fc)
+                                [&active_tokens](oneapi::tbb::flow_control&fc)
                                 {
                                     ++active_tokens;
                                     doWork();
                                     CHECK_MESSAGE(active_tokens <= n_tokens, "max number of tokens is exceed");
                                     --active_tokens;
-                                    if(++counter>=max_counter)
-                                    {
+                                    if (--input_counter < 0) {
```

Comment:
why not make existing variable `counter` an `atomic` and avoid using global variables ?

Reply:
it seems to be the common approach across all test cases.